### PR TITLE
Mybatis generator dynamic sql

### PR DESCRIPTION
**Pull request by bbaayezi**

### CODE REVIEW

File: src/main/java/com/ucl/imaginethisserver/Conf/GsonConfig.java [hunk 1]
```diff
@@ -20,11 +23,29 @@ public Gson gson() {
...
+
     public static class SwaggerJsonTypeAdapter implements JsonSerializer<Json> {
 
         @Override
         public JsonElement serialize(Json json, Type type, JsonSerializationContext context) {
             return JsonParser.parseString(json.value());
         }
     }
+
+//    private GsonHttpMessageConverter customGsonHttpMessageConverter() {
```
Comment (lukasotocerny) [inline src/main/java/com/ucl/imaginethisserver/Conf/GsonConfig.java:41]: I'd remove ignored code, or add comment saying why it might be used in the future

File: src/main/java/com/ucl/imaginethisserver/Service/ServiceImpl/VoteServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.ucl.imaginethisserver.Service.ServiceImpl;
+
+public class VoteServiceImpl {
+}
```
Comment (lukasotocerny) [inline src/main/java/com/ucl/imaginethisserver/Service/ServiceImpl/VoteServiceImpl.java:4]: Why this isn't implemented?

**Review by lukasotocerny — CHANGES_REQUESTED** [review]
Just 2 comments. And make sure to merge master into your branch, I believe you will have to delete some unit tests.

File: src/main/java/com/ucl/imaginethisserver/Service/ServiceImpl/VoteServiceImpl.java (hunk 2 above)
Comment (bbaayezi) [reply src/main/java/com/ucl/imaginethisserver/Service/ServiceImpl/VoteServiceImpl.java:4]: the voting functions are implemented in the feedbackServiceImpl, is it better to move them here instead?

**Comment by bbaayezi** [comment]
@lukasotocerny I tried to merge the master branch to the current branch, and I've found that unfortunately the old FeedbackServiceImplTest no longer works.
The thing is, in our previous design, the DAO classes basically have all the methods defined in the service classes, and the DAO classes were handling the actual database operations. In this branch, however, mybatis-generator were used to generate `Mapper`, where typical database operations were written in a `dynamic-sql` style. For example: 
```
feedbackMapper.select(c -> c
                .where(FeedbackDynamicSqlSupport.projectId, isEqualToWhenPresent(projectID))
        )
```
In short, the mappers wrapped all basic `CURD` methods like `insert` or `delete` for you and you are free to build up complex SQL statements. In our case, the mappers were used directly in the ServiceImpl classes, and the DAO classes were used to map more complex hand-written SQLs using annotations.

So now in this branch, the ServiceImpl classes also involve database operations, whose responsibility was taken by the DAO classes in our previous implementations. I don't plan to test the mappers since these methods were generated by the mybatis-generator. The new tests for ServiceImpl should look like DAO tests.

**Comment by bbaayezi** [comment]
Or do you want me to populate DAO classes like what we did before? I will only use mappers in DAO classes and in ServiceImpl there will only include business logics like check if the payload of the request body is valid.

**Review by lukasotocerny — APPROVED** [review]
I think it looks great 👍 LGTM

**Review by lukasotocerny — APPROVED** [review]
LGTM
